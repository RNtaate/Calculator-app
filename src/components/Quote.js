import React from 'react';

let Quote = () => {
  return (
    <p className="quote-par" style={{
      fontSize: '25px',
      padding: '20px 100px'
    }}>Mathematics is not about numbers, equations, computations,
    or algorithms: it is about understanding. â€“William Paul Thurston</p>
  );
}

export default Quote;