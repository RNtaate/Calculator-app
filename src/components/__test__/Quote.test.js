import React from 'react';
import ReactDOM from 'react-dom';
import Quote from '../Quote';

import {render} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import renderer from 'react-test-renderer';

it('Renders without crashing', () => {
  const myDiv = document.createElement('div');

  ReactDOM.render(<Quote />, myDiv);
});

it('renders the quote correctly', () => {
  const {getByTestId} = render(<Quote myQuote="This is passed down"/>);
  expect(getByTestId('quote')).toHaveTextContent('This is passed down');
});

it('renders the quote correctly', () => {
  const {getByTestId} = render(<Quote />);
  expect(getByTestId('quote')).toHaveTextContent('Mathematics is not about numbers, equations, computations,or algorithms: it is about understanding. â€“William Paul Thurston');
});

it("matches snapshot", () => {
  const tree = renderer.create(<Quote myQuote="This will be passed to quote"/>);
  expect(tree).toMatchSnapshot();
});

it("matches snapshot", () => {
  const tree = renderer.create(<Quote />);
  expect(tree).toMatchSnapshot();
});