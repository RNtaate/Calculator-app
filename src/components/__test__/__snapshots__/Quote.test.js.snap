// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<p
  className="quote-par"
  data-testid="quote"
  style={
    Object {
      "fontSize": "25px",
      "padding": "20px 100px",
    }
  }
>
  This will be passed to quote
</p>
`;

exports[`matches snapshot 2`] = `
<p
  className="quote-par"
  data-testid="quote"
  style={
    Object {
      "fontSize": "25px",
      "padding": "20px 100px",
    }
  }
>
  Mathematics is not about numbers, equations, computations,or algorithms: it is about understanding. â€“William Paul Thurston
</p>
`;
